SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPossession/SpatialTestRepossession.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3066/OwnerOnlyPropertyReplication.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3157/RPCInInterfaceTest.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestHandover/SpatialTestHandover.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/ReplicatedVisibilityTestActor.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/VisibilityTest/VisibilityTest.h"
+#include "Tests/AutomationEditorCommon.h"
+
+DEFINE_LOG_CATEGORY(LogGenerateTestMapsCommandlet);
+
+UGenerateTestMapsCommandlet::UGenerateTestMapsCommandlet()
+{
+	IsClient = false;
+	IsEditor = true;
+	IsServer = false;
+	LogToConsole = true;
+
+	ConstructorHelpers::FObjectFinder<UStaticMesh> PlaneAsset(TEXT("StaticMesh'/Engine/BasicShapes/Plane.Plane'"));
+	check(PlaneAsset.Succeeded());
+	PlaneStaticMesh = PlaneAsset.Object;
+
+	ConstructorHelpers::FObjectFinder<UMaterial> MaterialAsset(TEXT("Material'/Engine/BasicShapes/BasicShapeMaterial.BasicShapeMaterial'"));
+	check(MaterialAsset.Succeeded());
+	BasicShapeMaterial = MaterialAsset.Object;
+}
+
+UWorld* UGenerateTestMapsCommandlet::CreateNewTestMapWorld()
+{
+	UWorld* CurrentWorld = FAutomationEditorCommonUtils::CreateNewMap();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	AStaticMeshActor* Plane =
+		CastChecked<AStaticMeshActor>(GEditor->AddActor(CurrentLevel, AStaticMeshActor::StaticClass(), FTransform::Identity));
+	Plane->GetStaticMeshComponent()->SetStaticMesh(PlaneStaticMesh);
+	Plane->GetStaticMeshComponent()->SetMaterial(0, BasicShapeMaterial);
+	Plane->SetActorScale3D(FVector(
+		500, 500, 1)); // Make the initial platform larger so things don't fall off for tests which use a large area (visibility test)
+
+	AExponentialHeightFog* Fog =
+		CastChecked<AExponentialHeightFog>(GEditor->AddActor(CurrentLevel, AExponentialHeightFog::StaticClass(), FTransform::Identity));
+
+	APlayerStart* PlayerStart =
+		CastChecked<APlayerStart>(GEditor->AddActor(CurrentLevel, APlayerStart::StaticClass(), FTransform(FVector(-500, 0, 100))));
+
+	ASkyLight* SkyLight = CastChecked<ASkyLight>(GEditor->AddActor(CurrentLevel, ASkyLight::StaticClass(), FTransform::Identity));
+
+	// Use current settings for all these generated maps as default.
+	// This will probably change with UNR-4801
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->TestingSettings.TestingMode = EMapTestingMode::UseCurrentSettings;
+	WorldSettings->bEnableDebugInterface = true; // Setting to true for all test maps by default, TODO: consult Nic
```

Comment:
Please do TODO

Reply:
Good point. @ImprobableNic: do you think enabling the DebugInterface by default for tests would be detrimental? There is one test which queries the layered LB strategy directly, which does not work with the debug LB strat, but it's more of an exception.
What are your thoughts?